pkg/provider/server.go

```diff
@@ -129,18 +136,53 @@ func (m *SpringCloudConfigCSIProviderServer) Mount(ctx context.Context, req *v1a
 			Code: m.errorCode,
 		},
 	}
-	file, err := os.OpenFile(path.Join(req.GetTargetPath(), fileName), os.O_RDWR|os.O_CREATE, filePermission)
-	if err != nil {
-		return nil, fmt.Errorf("secrets store csi driver failed to mount %s at %s: %w", fileName, req.GetTargetPath(), err)
-	}
-	defer file.Close()
 
-	_, err = io.Copy(file, content)
-	if err != nil {
-		return nil, fmt.Errorf("secrets store csi driver failed to mount %s at %s: %w", fileName, req.GetTargetPath(), err)
+	if attrib.FileType != "" {
+		fileName := fmt.Sprintf("%s-%s.%s", attrib.Application, attrib.Profile, attrib.FileType)
+		content, err := m.springCloudConfigClient.GetConfig(attrib)
+		if err != nil {
+			return nil, fmt.Errorf("failed to retrieve secrets for %s: %w", fileName, err)
+		}
+		defer content.Close()
+
+		file, err := os.OpenFile(path.Join(req.GetTargetPath(), fileName), os.O_RDWR|os.O_CREATE, filePermission)
+		if err != nil {
+			return nil, fmt.Errorf("secrets store csi driver failed to mount %s at %s: %w", fileName, req.GetTargetPath(), err)
+		}
+		defer file.Close()
+
+		_, err = io.Copy(file, content)
+		if err != nil {
+			return nil, fmt.Errorf("secrets store csi driver failed to mount %s at %s: %w", fileName, req.GetTargetPath(), err)
+		}
+		log.Infof("secrets store csi driver mounted %s", fileName)
+		log.Infof("mount point: %s", req.GetTargetPath())
+	}
+
+	for idx, item := range attrib.Raw {
+		err = func() error {
```

Comment:
Instead of creating an anonymous function, consider putting the body of it into a dedicated function on `SpringCloudConfigCSIProviderServer`. Something like `mountRawFile()`. That allows for separation abd makes it easier to understand what the function is doing.